GliaWidgets/Sources/ViewModel/Chat/ChatViewModel.SecureConverstaions.swift

```diff
@@ -1,6 +1,25 @@
 extension ChatViewModel {
     func migrate(from transcript: SecureConversations.TranscriptModel) {
         sections = transcript.sections
+        // There's a possibility where migration to engagement
+        // happens when there are awaiting uploads.
+        // For that case we need to make sure that theses uploads
+        // have also migrated to use engagement based API.
+        fileUploadListModel.environment.uploader.uploads = transcript
+            .fileUploadListModel
+            .environment
+            .uploader
+            .uploads
+            .map { [environment] upload in
+                upload.environment = upload.environment.transformEnvironment(\.uploadFile) {
```

Comment:
I am not that familiar with this part of the code. How come there is a need to add this `transformEnvironment` method instead of assigning whatever new value directly?

Reply:
After spending some time in SwiftUI I thought it was necessary, but this is overkill here. So I remove it c21d0c39502db5f7d66d3fd8bed694bd10c17332.